pkg/ivd/ivd_protected_entity_type_manager.go

```diff
@@ -40,36 +41,47 @@ type IVDProtectedEntityTypeManager struct {
 	vsom      *vslm.GlobalObjectManager
 	cnsClient *cns.Client
 	s3URLBase string
-	user      string // These are being kept so we can open VDDK connections, may be able to open a VDDK connection
-	password  string // in IVDProtectedEntityTypeManager instead
+	vcParams  map[string]interface{} // Save the VC configuration params
 	logger    logrus.FieldLogger
 }
 
 func NewIVDProtectedEntityTypeManagerFromConfig(params map[string]interface{}, s3URLBase string,
 	logger logrus.FieldLogger) (*IVDProtectedEntityTypeManager, error) {
+	logger.Infof("Creating NewIVDProtectedEntityTypeManagerFromConfig. Params: %v", params)
 	var vcURL url.URL
-	vcHostStr, ok := params["vcHost"].(string)
+	vcHostStr, ok := params["VirtualCenter"].(string)
 	if !ok {
-		return nil, errors.New("Missing vcHost param, cannot initialize IVDProtectedEntityTypeManager")
+		return nil, errors.New("Missing VirtualCenter param, cannot initialize IVDProtectedEntityTypeManager")
 	}
+	vcHostPortStr, ok := params["port"].(string)
+	if !ok {
+		return nil, errors.New("Missing port param, cannot initialize IVD PETM")
+	}
+
 	vcURL.Scheme = "https"
-	vcURL.Host = vcHostStr
+	vcURL.Host = fmt.Sprintf("%s:%s", vcHostStr, vcHostPortStr)
 	insecure := false
-	insecureStr, ok := params["insecureVC"].(string)
-	if ok && (insecureStr == "Y" || insecureStr == "y") {
+	insecureStr, ok := params["insecure-flag"].(string)
+	if ok && (insecureStr == "TRUE" || insecureStr == "true" || insecureStr == "1" || insecureStr == "Y" || insecureStr == "y") {
 		insecure = true
 	}
-	vcUser, ok := params["vcUser"].(string)
+	vcUser, ok := params["user"].(string)
```

Comment:
Here and elsewhere, the preferred way in go would be 

`
if vcUser, ok := params["user"].(string); !ok {
    return nil, errors.New("Missing user param, cannot initialize IVDProtectedEntityTypeManager")
}
`

You could verify if this works and make changes if needed.

Reply:
+1

Reply:
Oh good to know. But I think it is a general coding style feedback and we might want to do it as a wider change. Is that ok?

Reply:
I tried this but got an error saying vcUser was unused.